### PR TITLE
Merge more symbolic meta kernels and symint changes from branch

### PR DESCRIPTION
**Pull request by bdhirsh**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86334

symintify split_with_sizes, dropout, fused_fake_obs_quant. meta for padding_2d ops

add meta_bernoulli_

meta kernel for at::gather

get pytorch_struct to pass: meta for scatter_add, fix backward

symintify split ops

### CODE REVIEW

**Review by ezyang — APPROVED** [review]
It would be nice if we had test_meta.py tests on register_dispatcher=False regs

**Comment by bdhirsh** [comment]
> It would be nice if we had test_meta.py tests on register_dispatcher=False regs

Coming up here :) https://github.com/pytorch/pytorch/pull/86228

A bunch of errors actually came out of it, so I'm starting to sift through them

File: torch/csrc/autograd/FunctionsManual.cpp [hunk 1]
```diff
@@ -1840,17 +1840,18 @@ Tensor split_with_sizes_backward(
...
+    c10::SymIntArrayRef sym_sizes,
     const at::TensorOptions& options) {
-  dim = at::maybe_wrap_dim(dim, sizes.size());
-  int64_t dim_size = sizes[dim];
+  dim = at::maybe_wrap_dim(dim, sym_sizes.size());
+  auto dim_size = sym_sizes[dim];
   int64_t num_splits = grads.size();
-  std::vector<int64_t> split_sizes(num_splits, split_size);
+  std::vector<c10::SymInt> split_sizes(num_splits, split_size);
+  auto sym_split_size = c10::SymInt(split_size);
```
Comment (albanD) [inline torch/csrc/autograd/FunctionsManual.cpp:1851]: It is already a SymInt

File: torch/_meta_registrations.py [hunk 2]
```diff
@@ -1472,18 +1472,25 @@ def meta_scatter_add(self, dim, index, src):
...
+        for i in range(spatial_dimensions):
+            sym_float = (input_shape[i + 2] / 1) * scale_factors[i]
+            assert sym_float >= 0
+            out_size.append(math.floor(sym_float))
+
+    output_height = out_size[0]
+    output_width = out_size[1]
+    nbatch = input_shape[0]
+    channels = input_shape[1]
+    return input.new_empty((nbatch, channels, output_height, output_width)).to(memory_format=mem_format)
```
Comment (bdhirsh) [inline torch/_meta_registrations.py:1493]: cc @Krovatkin 

tbh I just started at the C++ meta kernel defined in `UpSampleNearest2d.cpp` to get this change, and it passed that failing `test/test_meta.py` test

**Comment by albanD** [comment]
reland is at https://github.com/pytorch/pytorch/pull/86488

**Pull request closed without merging**